doc/kuksa_requirementes.md

```diff
@@ -0,0 +1,805 @@
+    - [Terminology](#terminology)
+    - [System Architecture](#system-architecture)
+- [Functional requirements](#functional-requirements)
+    - [As Client](#as-client)
+    - [As Databroker](#as-databroker)
+    - [As Provider](#as-provider)
+- [Non-Functional Requirements](#non-functional-requirements)
+  - [Data availability/persistence according to lifecycle of Client, Databroker and Provider](#data-availabilitypersistence-according-to-lifecycle-of-client-databroker-and-provider)
+  - [Wildcard support](#wildcard-support)
+  - [Registration of Datapoints](#registration-of-datapoints)
+  - [Availability of Datapoints](#availability-of-datapoints)
+  - [Lifecycle of components](#lifecycle-of-components)
+  - [Path requests](#path-requests)
+  - [Errors](#errors)
+  - [Setting Values](#setting-values)
+  - [Atomic operatoins](#atomic-operatoins)
+  - [Update notifications](#update-notifications)
+  - [Access rights](#access-rights)
+  - [VSS signals - Users vs providers](#vss-signals---users-vs-providers)
+  - [Enable "easy to use" user facing API](#enable-easy-to-use-user-facing-api)
+  - [Performance / runtime footprint](#performance--runtime-footprint)
+  - [Throttling-mode when system is overloaded.](#throttling-mode-when-system-is-overloaded)
+  - [Considerations regarding shared and zero-copy memory approaches](#considerations-regarding-shared-and-zero-copy-memory-approaches)
+  - [Provider control and provider capabilities](#provider-control-and-provider-capabilities)
+  - [Control the rate of updates](#control-the-rate-of-updates)
+  - [Differentiate between different providers of the same VSS data](#differentiate-between-different-providers-of-the-same-vss-data)
+  - [Data Aliveness/Availability](#data-alivenessavailability)
+  - [Missing features from `sdv.databroker.v1` in `kuksa.val.v1`](#missing-features-from-sdvdatabrokerv1-in-kuksavalv1)
+- [Exploring a design of a bidirectional streaming API](#exploring-a-design-of-a-bidirectional-streaming-api)
+    - [Design choices](#design-choices)
+        - [Stream](#stream)
+        - [Bidirectional stream](#bidirectional-stream)
+        - [Actuators](#actuators)
+    - [Overview](#overview)
+  - [COVESA topics](#covesa-topics)
+- [Domain requirements](#domain-requirements)
+  - [Interface requirements](#interface-requirements)
+    - [Current Databroker API design](#current-databroker-api-design)
+    - [New Databroker API proposal](#new-databroker-api-proposal)
+- [Appendix](#appendix)
+  - [1. Use Cases](#1-use-cases)
+  - [2. Protobuf API Diagram](#2-protobuf-api-diagram)
+
+# Documentation
+  ### [Terminology](./terminology.md)
+  ### [System Architecture](./system-architecture.md)
+
+# Functional requirements
+
+### As Signal Consumer
+1. As a client I want to have a single service point accessible from my execution environment.
+   
+2. As a client I want to `Get` sensor values or actuator current values of the vehicle and its environment in a uniform manner.
```

Comment:
how about `of the vehicle signals` instead of `vehicle and its environment` because I am not sure how to re-define vehicle and its environment when we already have definition for signal.

Reply:
In my initial document this was part of a "basic requirement". Signals of the vehicle's environment would be outside air temperature, for example

Reply:
Then I interpreted the `environment` too wide. I was more thinking about the information about traffic lights, charging infrastructure, attached trailers etc.